build_runner/lib/src/daemon/daemon_builder.dart

```diff
@@ -161,7 +161,9 @@ class BuildRunnerDaemonBuilder implements DaemonBuilder {
               expectedDeletes,
             ));
 
-    return BuildRunnerDaemonBuilder._(
-        builder, buildOptions, outputStreamController, graphEvents);
+    var changes =
+        graphEvents.map((data) => WatchEvent(data.type, data.id.path));
```

Comment:
I should have noticed this before - this won't work when you have changes in the non-root package with a path dependency. This looks like it's assuming all changes are in the root package only.

Reply:
I am now serializing the AssetId. This shouldn't be a problem as we are consuming the watch events. It does make some of the blacklist patterns more difficult to write but for the external world I don't think those will be used very often.